CA.md

```diff
@@ -0,0 +1,91 @@
+# Example CA Usage
+
+Although there is more than one way to handle extensions (grants/identities) and
+signing certificates, HIBA includes a `hiba-ca.sh` tool and `hiba-gen` binary.
+The following serves as a simple example of a way to get started with HIBA.
+
+Once you have completed the
+[compilation instructions](https://github.com/google/hiba/blob/main/INSTALL.md#compilation)
+you will have a `hiba-gen` and `hiba-ca.sh` in your hiba directory. You can now
+create CA keys, grants, identities, grant eligibility, and sign host and user
+certificates with these extensions. The following assumes you are in your hiba
+directory. The CA structure will be in `~/.hiba-ca` by default.
+
+1. Generate a CA keypair.
+    ```
+    $ ./hiba-ca.sh -c
+    == Generating CA keys in ~/.hiba-ca/ca
+    Enter passphrase (empty for no passphrase):
+    Enter same passphrase again:
+    == Done ==
+    ```
+1. Generate a user keypair
+    ```
+    $ ./hiba-ca.sh -c -u -I testuser
+    == Generating user keys in ~/.hiba-ca/users/testuser
+    == Done ==
+    Identity created: Private: ~/.hiba-ca/users/testuser
+                      Public:  ~/.hiba-ca/users/testuser.pub
+    ```
+    * Alternatively you can copy existing keys
+        ```
+        $ ./hiba-ca.sh -i -u -f ~/.ssh/id_rsa.pub -I testuser
+        == Done ==
+        Identity imported: Public: ~/.hiba-ca/users/testuser.pub
+        ```
+1. Copy the OpenSSH server host public key from the OpenSSH server to your
+   machine and import it.
+    ```
+    ./hiba-ca.sh -i -h -f path_to_pubkey -I hostname
+    ```
+    * Alternatively you can generate a new host keypair
+        ```
+		./hiba-ca.sh -c -h -I hostname
+		== Generating host keys in ~/.hiba-ca/hosts/hostname
+		Enter passphrase (empty for no passphrase):
+		Enter same passphrase again:
+		== Done ==
+		Identity created: Private: ~/.hiba-ca/hosts/hostname
+						  Public:  ~/.hiba-ca/hosts/hostname.pub
+        ```
+1. Create an identity for prod, with a domain of example.com.
+    ```
+    $ ./hiba-gen -i -f "~/.hiba-ca/policy/identities/prod" domain example.com
+    ```
+1. Create a grant called shell with a matching domain.
```

Comment:
nit: maybe put in backquotes?

Create a grant called `shell` with a matching domain